### ISSUE TITLE
rowPadding mostly applies to bottom?

### ISSUE THREAD

**Issue by courajs** [jasonmit/virtual-each#99]
I'm not 100% on this, but I'm noticing un-rendered white flashes when scrolling upward, and increasing `rowPadding` doesn't seem to help. It does seem to help with white flashes when scrolling downward.

So, my guess is that the extra rows are only being rendered below the viewport, not above.

I had some trouble following all the math in `virtual-each/component.js`, but I got the impression that my guess is right.

Would you be open to rendering the extra rows _around_ the viewport instead? It will introduce some more complexity to the scroll state around the top/bottom of the list.

**Comment by jasonmit** [comment]
Unsure I follow adding padding around viewport.  Could you screen grab what you are seeing?  (http://www.cockos.com/licecap/ If you need an app for this)

**Comment by courajs** [comment]
Sure. I'm specifying a `rowHeight` of 10 in the component:
```hbs
  {{#virtual-each options
    height=containerHeight
    itemHeight=itemHeight
    rowPadding=10
    as |item actualIndex virtualIndex|
  }}
    ...
  {{/virtual-each}}
```
![rowpadding](https://cloud.githubusercontent.com/assets/1588273/24967664/dd6d9b16-1f78-11e7-81f8-2ca812491bf8.gif)

As you can see, the 10 `rowPadding` rows are rendered below the viewport, so scrolling down doesn't show a white flash. But no extra rows are rendered above the viewport, so scrolling up shows a white flash.

A `rowPadding` of 10 should probably render 5 above and 5 below, or 10 above and 10 below, rather than just 10 below.

**Comment by jasonmit** [comment]
Awesome thanks, I may need more info once I start looking into this.  Need to jump on a flight but I'll have a look over the next few days, possibly tonight.

**Comment by jasonmit** [comment]
> should probably render 5 above and 5 below, or 10 above and 10 below, rather than just 10 below

Agreed.  The algorithm should be adjusted to pad the top and bottom equally instead of just the bottom.

**Comment by jasonmit** [comment]
Agreed, I'll start thinking about a fix since the algorithm will need to be changed around a bit to support this.
